tests/qunit/core/common/assets/js/api/core/data.spec.js

```diff
@@ -0,0 +1,439 @@
+				args = { query: {} };
+
+			args.query.paramA = 'valueA';
+			args.query.paramB = 'valueB';
+
+			const endpoint = $e.data.commandToEndpoint( command, args );
+
+			assert.equal( endpoint, 'component?paramA=valueA&paramB=valueB&', 'Valid endpoint.' );
+		} );
+
+		QUnit.test( 'endpointToCommand(): endpoints with index and id', ( assert ) => {
+			const Component = class extends ComponentBase {
+					getNamespace() {
+						return 'test-data-commands';
+					}
+
+					defaultData() {
+						return {
+							index: () => {},
+							'test-id': () => {},
+						};
+					}
+				},
+				component = $e.components.register( new Component() );
+
+			const endpointWithIndex = $e.data.endpointToCommand( 'test-data-commands', {} ),
+				endpointWithId = $e.data.endpointToCommand( 'test-data-commands/test-id/some-id', {} );
+
+			assert.equal( endpointWithIndex, component.getNamespace() + '/index' );
+			assert.equal( endpointWithId, component.getNamespace() + '/test-id' );
+		} );
+
+		QUnit.test( 'prepareHeaders(): with GET', ( assert ) => {
+			const requestData = { paramA: 'valueA' },
+				params = $e.data.prepareHeaders( 'get', requestData );
+
+			assert.equal( params.headers[ 'X-WP-Nonce' ], wpApiSettings.nonce );
+		} );
+
+		QUnit.test( 'prepareHeaders(): with POST', ( assert ) => {
+			const requestData = { paramA: 'valueA' },
+				params = $e.data.prepareHeaders( 'create', requestData );
+
+			assert.equal( params.body, '{"paramA":"valueA"}' );
+		} );
+
+		QUnit.test( 'prepareHeaders(): with invalid type', ( assert ) => {
+			const type = 'some-invalid-type';
+
+			assert.throws( () => $e.data.prepareHeaders( type, {} ),
+				new Error( `Invalid type: '${ type }'` )
+			);
+		} );
+
+		/**
+		 * FETCH, tests: is not mean to test if the *cache* mechanism is working correctly, but
+		 * to test *fetch* with different approaches.
+		 */
+		QUnit.test( 'fetch(): simple', async ( assert ) => {
+			const testComponent = $e.components.register( new class TestComponent extends ComponentBase {
+				getNamespace() {
+					return 'test-component-fetch-simple';
+				}
+
+				defaultData() {
+					return this.importCommands( {
+						TestCommand: class TestCommand extends CommandData {},
+					} );
+				}
+			} );
+
+			const fetchOrig = fetch,
+				command = 'test-component-fetch-simple/test-command',
+				response = {
+					json: () => response,
+					ok: true,
+					data: {
+						testParam: 'testValue',
+					},
+				},
+				args = {
+					options: {
+						refresh: true,
+					},
+					query: {
+						param1: 'valueA',
+					},
+				},
+				endpoint = $e.data.commandToEndpoint( command, args );
+
+			fetch = ( address, params ) => {
+				fetch = fetchOrig;
+				return new Promise( async ( resolve ) => {
+					resolve( response );
+				} );
+			};
+
+			const result = await $e.data.fetch( 'get', {
+				endpoint,
+				command,
+				args,
+			} );
+
+			// Validate result.
+			assert.deepEqual( response.data, result.data );
```

Comment:
How the `fetch` knows to return the `testParam`?

Reply:
Rewrite test.

Reply:
Handled.